csharp/src/Apache.Arrow.Flight/FlightRecordBatchStreamWriter.cs

```diff
@@ -64,7 +64,7 @@ protected virtual void Dispose(bool disposing)
         {
             if (!_disposed)
             {
-                _flightDataStream.Dispose();
+                _flightDataStream?.Dispose();
```

Comment:
This isn't required for any of the included tests, but before disabling the `primitive_no_batches` test, it would crash here with a NullReferenceException due to the writer being disposed before the stream was created.

With this fix, writing doesn't crash, but the data isn't found when trying to retrieve it.